framework/src/modules/interoperability/sidechain/commands/mainchain_registration.ts

```diff
@@ -76,10 +80,21 @@ export class MainchainRegistrationCommand extends BaseInteroperabilityCommand {
 			};
 		}
 
+		const interoperabilityStore = this.getInteroperabilityStore(context);
+		const ownChainAccount = await interoperabilityStore.getOwnChainAccount();
+		if (!ownChainID.equals(ownChainAccount.id)) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(`Invalid ownChainID property.`),
+			};
+		}
+
 		if (!isValidName(ownName)) {
 			return {
 				status: VerifyStatus.FAIL,
-				error: new Error(`Sidechain name is in an unsupported format: ${ownName}`),
+				error: new Error(
+					`Invalid ownName property. It should contain only characters from the set [a-z0-9!@$&_.].`,
+				),
 			};
 		}
 
```

Comment:
Instead of using `NUMBER_MAINCHAIN_VALIDATORS`, probably better to use `mainchainValidators.length`

Reply:
do you think there can be a case when it's not the same ? I am saying because it is verified in the schema `minItems: NUMBER_MAINCHAIN_VALIDATORS,
			maxItems: NUMBER_MAINCHAIN_VALIDATORS,`